projects/gloo/cli/pkg/cmd/install/uninstall.go

```diff
@@ -90,15 +88,18 @@ func deleteNamespace(cli install.KubeCli, namespace string) error {
 }
 
 func uninstallKnativeIfNecessary(cli install.KubeCli) error {
-	installClient := DefaultGlooKubeInstallClient{}
-	knativeExists, isOurInstall, err := installClient.CheckKnativeInstallation()
+	_, installOpts, err := checkKnativeInstallation()
 	if err != nil {
 		return errors.Wrapf(err, "finding knative installation")
 	}
-	if knativeExists && isOurInstall {
-		fmt.Printf("Removing namespace %s...\n", constants.KnativeServingNamespace)
-		if err := cli.Kubectl(nil, "delete", "namespace", constants.KnativeServingNamespace); err != nil {
-			return errors.Wrapf(err, "delete knative failed")
+	if installOpts != nil {
+		fmt.Printf("Removing installed knative components %#v...\n", installOpts)
```

Comment:
Maybe make it clearer that we are removing stuff that was installed by us?
```suggestion
		fmt.Printf("Removing knative components installed by Gloo %#v...\n", installOpts)
```